File: middleware/src/dsl/Battery.java
package dsl;

public class Battery extends EnergyResource {
	private int totalEnergy;
	private int currentEnergy;
	
	public Battery(int energy) {
		this.totalEnergy = energy;
	}
	
	public int getTotalEnergy() {
		return totalEnergy;
	}
	
	public boolean isEnergyDeplenished() {
		return currentEnergy == 0;
	}
	
	public void spendEnergy(int e) {
		currentEnergy -= e;
		if (currentEnergy < 0) {
			currentEnergy = 0;
		}
	}
	
	public double getRemainingEnergyPercentage() {
		return ((double)currentEnergy) / ((double)totalEnergy)
	}
}

File: middleware/src/dsl/mission/Battery.py
from .EnergyResource import EnergyResource

class Battery(EnergyResource):
	totalEnergy = 0
	currentEnergy = 0
	
	def __init__(self, energy):
		self.totalEnergy = energy
	
	def getTotalEnergy(self):
		return self.totalEnergy
	
	def isEnergyDeplenished(self):
		return self.currentEnergy == 0
	
	def spendEnergy(self, e):
		self.currentEnergy -= e
		if self.currentEnergy < 0:
			currentEnergy = 0

	def getRemainingEnergyPercentage(self):
		return (self.currentEnergy) / (self.totalEnergy)